src/main/java/org/jenkinsci/plugins/workflow/cps/actions/ArgumentsActionImpl.java

```diff
@@ -0,0 +1,197 @@
+            "CHANGE_ID",
+            "CHANGE_TARGET",
+            "CHANGE_TITLE",
+            "CHANGE_URL",
+            "EXECUTOR_NUMBER",
+            "HUDSON_COOKIE",
+            "HUDSON_HOME",
+            "HUDSON_SERVER_COOKIE",
+            "HUDSON_URL",
+            "JENKINS_HOME",
+            "JENKINS_SERVER_COOKIE",
+            "JENKINS_URL",
+            "JOB_BASE_NAME",
+            "JOB_NAME",
+            "JOB_URL",
+            "NODE_LABELS",
+            "NODE_NAME",
+            "WORKSPACE",
+
+            // Normal system variables for posix environments
+            "HOME",
+            "LANG",
+            "LOGNAME",
+            "MAIL",
+            "NLSPATH",
+            "PATH",
+            "PWD",
+            "SHELL",
+            "SHLVL",
+            "TERM",
+            "USER",
+            "XFILESEARCHPATH"
+    ));
+
+    /**
+     * Does first-level sanitization of a step's params, returning the Describable arguments map
+     * @param stepArguments Arguments provided when creating the step
+     * @param variables Environment variables to remove
+     * @return
+     */
+    @Nonnull
+    public static Map<String,Object> sanitizedStepArguments(@Nonnull Map<String, Object> stepArguments, @CheckForNull EnvVars variables) {
+        HashMap<String, Object> output = new HashMap<String, Object>();
+
+        if (variables == null || variables.size() == 0) {
+            // No need to sanitize against environment variables
+            for (Map.Entry<String,Object> entry : stepArguments.entrySet()) {
+                if (entry.getValue() instanceof CharSequence && ((CharSequence)(entry.getValue())).length() > ArgumentsAction.MAX_STRING_LENGTH) {
+                    output.put(entry.getKey(), OVERSIZE_VALUE);
+                } else {
+                    output.put(entry.getKey(), entry.getValue());
+                }
+            }
+            return output;
+        }
+
+        for (Map.Entry<String,?> param : stepArguments.entrySet()) {
+            Object val = param.getValue();
+            if (val != null && val instanceof String) {
+                String valString = (String)val;
+                if (valString.length() > ArgumentsAction.MAX_STRING_LENGTH) {
+                    output.put(param.getKey(), NotStoredReason.OVERSIZE_VALUE);
+                } else {
+                    boolean isSafe = isStringSafe(valString, variables, SAFE_ENVIRONMENT_VARIABLES);
+                    if (isSafe) {
+                        output.put(param.getKey(), valString);
+                    } else {
+                        output.put(param.getKey(), MASKED_VALUE);
+                    }
+                }
+            } else {
+                output.put(param.getKey(), val);
+            }
+        }
+        return output;
+    }
+
+    @Nonnull
+    @Override
+    protected Map<String, Object> getArgumentsInternal() {
+        return (arguments == null)? Collections.EMPTY_MAP : arguments;
```

Comment:
Another tiny nit - parens are not needed here, would love a space before the `?`, but yeah, obviously not a blocker.